tests/unit/conftest.py

```diff
@@ -95,3 +95,13 @@ def mock_certificate_fixture() -> str:
         "4+3+0/6Ba2Zlt9fu4PixG+XukQnBIxtIMjWp7q7xWp8F4aOW"
         "-----END CERTIFICATE-----"
     )
+    provider_cert_mock = MagicMock()
```

Comment:
consider using the `spec` attribute for MagicMock

Reply:
I rephrased this to be more explicit.